lib/qif/writer.rb

```diff
@@ -61,11 +62,17 @@ def initialize(io, type = 'Bank', format = 'dd/mm/yyyy')
     
     # Add a transaction for writing
     def <<(transaction)
-      @transactions << transaction
+      case transaction.class.to_s
```

Comment:
@davidsklar did say that his !Account was found at the end of the file. But if !Account is at the start of a transaction block something like this should work without too much of a refactor.

```ruby
@accounts = {}

def <<(transaction)
  case transaction.class.name 
  when 'Qif::Account'     then @account = transaction
  when 'Qif::Transaction' then @accounts[@account.to_s] << transaction
  else raise "Cannot handle #{transaction.class.name.inspect}"
  end
end

def write
  @accounts.each do |account,transactions|
    write_record account
    write_header
    transactions.each {|t| write_transaction }
  end
end
```